vulnerabilities/importers/mozilla.py

```diff
@@ -0,0 +1,180 @@
+    """
+    Convert a file to corresponding advisories.
+    This calls proper method to handle yml/md files.
+    """
+    mfsa_id = mfsa_id_from_filename(path)
+    if not mfsa_id:
+        return []
+
+    with open(path) as lines:
+        if path.endswith(".md"):
+            return get_advisories_from_md(mfsa_id, lines)
+        if path.endswith(".yml"):
+            return get_advisories_from_yml(mfsa_id, lines)
+
+    return []
+
+
+def get_advisories_from_yml(mfsa_id, lines) -> List[Advisory]:
+    advisories = []
+    data = yaml.safe_load(lines)
+    data["mfsa_id"] = mfsa_id
+
+    fixed_package_urls = get_package_urls(data.get("fixed_in"))
+    references = get_yml_references(data)
+
+    if not data.get("advisories"):
+        return []
+
+    for cve, advisory in data["advisories"].items():
+        # These may contain HTML tags
+        summary = BeautifulSoup(advisory.get("description", ""), features="lxml").get_text()
+
+        advisories.append(
+            Advisory(
+                summary=summary,
+                vulnerability_id=cve if is_cve(cve) else "",
+                impacted_package_urls=[],
+                resolved_package_urls=fixed_package_urls,
+                references=references,
+            )
+        )
+
+    return advisories
+
+
+def get_advisories_from_md(mfsa_id, lines) -> List[Advisory]:
+    yamltext, mdtext = split_markdown_front_matter(lines.read())
+    data = yaml.safe_load(yamltext)
+    data["mfsa_id"] = mfsa_id
+
+    fixed_package_urls = get_package_urls(data.get("fixed_in"))
+    references = get_yml_references(data)
+    cves = re.findall(r"CVE-\d+-\d+", yamltext + mdtext, re.IGNORECASE)
+    for cve in cves:
+        references.append(
+            Reference(
+                reference_id=cve,
+                url=f"https://cve.mitre.org/cgi-bin/cvename.cgi?name={cve}",
+            )
+        )
+
+    description = html_get_p_under_h3(markdown(mdtext), "description")
+
+    return [
+        Advisory(
+            summary=description,
+            vulnerability_id="",
+            impacted_package_urls=[],
+            resolved_package_urls=fixed_package_urls,
+            references=references,
+        )
+    ]
+
+
+def html_get_p_under_h3(html, h3: str):
+    soup = BeautifulSoup(html, features="lxml")
+    h3tag = soup.find("h3", text=lambda txt: txt.lower() == h3)
+    p = ""
+    if h3tag:
+        for tag in h3tag.next_siblings:
+            if tag.name:
+                if tag.name != "p":
+                    break
+                p += tag.get_text()
+    return p
+
+
+def mfsa_id_from_filename(filename):
+    match = MFSA_FILENAME_RE.search(filename)
+    if match:
+        return "mfsa" + match.group(1)
+
+    return None
+
+
+def get_package_urls(pkgs: List[str]) -> List[PackageURL]:
+    package_urls = [
+        PackageURL(
+            type="mozilla",
+            # pkg is of the form "Firefox ESR 1.21" or "Thunderbird 2.21"
+            name=pkg.rsplit(None, 1)[0],
```

Comment:
this split and the one below are rather cryptic, magic and likely error prone, please use a plain loop rather than a list comprehension and try to find a more readable and explicit way to do things. partition and rpartition are usually more robust and easier to grok.

Also avoid doing the multiple time the same op (here for the name and version)

Reply:
I've updated it to use a for loop. I've used rsplit to be consistent with the upstream. 
Here's what the upstream uses: https://github.com/mozilla/bedrock/blob/bd0f50a1c0a7455115cc2bf9eb3ae9027586a024/bedrock/security/models.py#L28-L30
It is mentioned in the commit message here https://github.com/nexB/vulnerablecode/pull/393/commits/05d93b2e3a88b15bceb5675b7cb46569d3fc216b

Reply:
rpartition is a more robust splitter than rsplit. Your rsplit will fail when there is no spaces 
```
>>> a, b = 'foo'.rsplit(None, 1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: not enough values to unpack (expected 2, got 1)
>>> a, _, b = 'foo'.rpartition(' ')

```

Reply:
If there's no space, `rpartition` puts the entire string on `b` which would be `version` here. imo, if `pkg="Mozilla"` it refers to the `name` and not the `version`.
To account for that, I came up with this. Not sure if it looks hacky.
```py
            name, _, version = pkg.rpartition(' ')
            if not name:
                name = version
```

Reply:
this works fine. Or you could do this slightly more explicit:
```
>>> def name_ver(s):
...     if ' ' in s:
...         name, _, ver = s.rpartition(' ')
...     else:
...         name = s
...         ver = None
...     return name, ver
... 
>>> name_ver('foo')
('foo', None)
>>> name_ver('foo 23.34')
('foo', '23.34')
```